main.py

```diff
@@ -45,17 +47,47 @@
             Select BOOLEAN such that Calls(_, _)
             """
         )
-
-        print(evaluate_query(tree, queries[0]))
+        context = extract(tree)
+        print(evaluate_query(queries[0], context))
 
     elif len(sys.argv) > 0:
+
+        def no_time_left():
+            print("\nPreparationTimeout")
+            exit(1)
+
         with open(sys.argv[-1]) as f:
             code = f.read()
             tree = parse(code)
-            # TODO: Extract design patterns here!
+            context = extract(tree)
             print("Ready")
+            while True:
+                try:
+                    t = Timer(60, no_time_left)
```

Comment:
Kiedy to ma sprawdzac?

Reply:
Aż nie ma zapytań query otrzymuje dwie puste linijki